### PR TITLE
feat: add a basic Cypress TypeScript example using ENV variables

### PR DESCRIPTION
**Pull request by Zidious**
This PR adds a basic Cypress TypeScript example using `env` variables both within the Cypress config and `cyress.env.json` file. 

No QA Required

### CODE REVIEW

File: cypress/basic-typescript/cypress.config.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { defineConfig } from 'cypress'
+import { cypressConfig } from '@axe-core/watcher'
+
+const { API_KEY, SERVER_URL = 'https://axe.deque.com' } = process.env
```
Comment (stephenmathieson) [inline cypress/basic-typescript/cypress.config.ts:4]: Can this be read with `Cypress.env`?

File: cypress/basic-typescript/cypress.config.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export default defineConfig(
+  cypressConfig({
+    axe: {
+      apiKey: API_KEY,
+      serverURL: SERVER_URL
+    },
+    env: {
+      password: 'SuperSecretPassword!'
```
Comment (stephenmathieson) [inline cypress/basic-typescript/cypress.config.ts:13]: Why is this specified here but the username in `cypress.env.json`?

File: cypress/basic-typescript/cypress/e2e/login.cy.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+describe('My Login Application', () => {
+  it('should login with valid credentials', () => {
+    const loginCredentials = Cypress.env('login')
```
Comment (stephenmathieson) [inline cypress/basic-typescript/cypress/e2e/login.cy.ts:3]: IMO this should be consistent. Either:

```suggestion
    const username = Cypress.env('login').username
    const password = Cypress.env('password')
```

or

```js
.type(Cypress.env('login').username)
.get('#password')
.type(Cypress.env('password'))
```

Comment (stephenmathieson) [inline cypress/basic-typescript/cypress/e2e/login.cy.ts:3]: Also an option (probably preferred):

```suggestion
    const { username, password } = Cypress.env('login')
```

**Review by stephenmathieson — COMMENTED** [review]
The example is not just "basic typescript", but "typescript with environment variables". Maybe name the directory `cypress/typescript-with-env`?

Also small nits left inline

**Comment by Zidious** [comment]
No Longer Required

**Pull request closed without merging**